idaes/core/base/tests/test_flowsheet_model.py

```diff
@@ -400,6 +400,39 @@ def test_dynamic_external_time(self):
         assert m.fs.sub.config.time is m.s
         assert m.fs.sub.time_units == units.minute
 
+    @pytest.mark.unit
+    def test_dynamic_nested_time(self):
+        m = ConcreteModel()
+        m.s1 = ContinuousSet(initialize=[4, 5])
+        m.s2 = ContinuousSet(initialize=[1,2,3])
+        m.fs = FlowsheetBlock(dynamic=True, time_units=units.s)
+        m.fs.sub = FlowsheetBlock(dynamic=True, time=m.s2, time_units=units.s)
+
+        assert m.fs.sub.config.dynamic is True
+        assert isinstance(m.fs.sub.time, ContinuousSet)
```

Comment:
Here you should do `assert m.fs.time is m.s2`. - i.e. check that the time domain is the correct set, not just the correct type.